src/components/ColorPickerContainer.ts

```diff
@@ -0,0 +1,269 @@
+        let errorMessage = "";
+        if (props.onChangeEvent === "callMicroflow" && !props.onChangeMicroflow) {
+            errorMessage = "on change event is set to 'Call a microflow' but no microflow is selected";
+        } else if (props.onChangeEvent === "showPage" && !props.onChangePage) {
+            errorMessage = "on change event is set to 'Show a page' but no page is selected";
+        } else if (props.onChangeEvent === "callNanoflow" && (JSON.stringify(props.onChangeNanoflow) === JSON.stringify({}))) {
+            errorMessage = "on change event is set to 'Call a nanoflow' but no nanoflow is selected";
+        }
+
+        return errorMessage && `Error in color picker configuration: ${errorMessage}`;
+    }
+
+    public static parseStyle(style = ""): {[key: string]: string} {
+        try {
+            return style.split(";").reduce<{[key: string]: string}>((styleObject, line) => {
+                const pair = line.split(":");
+                if (pair.length === 2) {
+                    const name = pair[0].trim().replace(/(-.)/g, match => match[1].toUpperCase());
+                    styleObject[name] = pair[1].trim();
+                }
+                return styleObject;
+            }, {});
+        } catch (error) {
+            // tslint:disable-next-line no-console
+            console.log("Failed to parse style", style, error);
+        }
+
+        return {};
+    }
+
+    private renderColorPicker(hasLabel = false) {
+        const alertMessage = this.state.alertMessage || ColorPickerContainer.validateProps(this.props);
+
+        return createElement(ColorPicker, {
+            alertMessage,
+            className: !hasLabel ? this.props.class : undefined,
+            color: this.state.color,
+            disabled: this.disabled,
+            type: this.props.type,
+            mode: this.props.mode,
+            close: this.handleClose,
+            displayColorPicker: this.state.displayColorPicker,
+            style: !hasLabel ? ColorPickerContainer.parseStyle(this.props.style) : undefined,
+            onChange: this.updateColorValue,
+            onChangeComplete: this.handleOnChange
+        }, this.props.mode === "input"
+                ? createElement(Input, {
+                    disabled: this.disabled,
+                    color: this.state.color,
+                    onChange: this.handleInputChange
+                }, this.renderColorPickerButton())
+                : this.renderColorPickerButton()
+        );
+    }
+
+    private renderColorPickerButton() {
+        return createElement(Button, {
+            className: this.props.mode === "input" ? "widget-color-picker-input-inner" : "widget-color-picker-inner",
+            disabled: this.disabled,
+            mode: this.props.mode,
+            color: this.state.color,
+            ...(!this.disabled && this.props.mode !== "inline")
+                ? { onClick: this.handleClick }
```

Comment:
Move to single line... 153... then move the conditional into `this.handleClick`

Reply:
Nice one..Done